docs/siem/detections/api/lists/api-delete-list-item.asciidoc

```diff
@@ -0,0 +1,30 @@
+[[lists-api-delete-item]]
+=== Delete list item
+
+Deletes a list item.
+
+==== Request URL
+
+`DELETE <kibana host>:<port>/api/lists/items`
+
+===== URL query parameters
+
+The URL query must include one of the following:
+
+* `id` - `DELETE /api/lists/items?id=<id>`
+* `list_id` and `value` - `DELETE /api/lists/items?list_id=<list_id>&value=<value>`
+
+===== Example request
+
+Deletes the list item with an `id` of `internal-ip-london`:
+
+[source,console]
+--------------------------------------------------
+DELETE api/lists/items?id=internal-ip-london
+--------------------------------------------------
+// KIBANA
+
+==== Response code
+
+`200`:: 
+    Indicates a successful call.
```

Comment:
One other feature not shown here within the API is a delete by value capability which works well with IP ranges and regular IP's.

For example if I import a set of ip's like I show above with an import I can actually delete a set of them using a CIDR query:

```ts
❯ ./delete_list_item_by_value.sh ips.txt 127.0.0.1/16
[
  {
    "created_at": "2020-08-04T20:35:55.194Z",
    "created_by": "yo",
    "id": "mbwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "47d3d57b-c5fe-4948-8d73-d147cfcf5c22",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.194Z",
    "updated_by": "yo",
    "value": "127.0.0.1"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "mrwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "e77c0764-98b8-4a6a-8a04-bb0bf62ff56c",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.2"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "m7wwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "4838d3d7-9958-4f97-8833-63fa07aa5a0b",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.3"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "nLwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "04e46f26-56dc-4d6a-8d65-e40725078183",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.4"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "nbwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "40f39f31-5c54-4e71-98ef-c2808836bc1b",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.5"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "nrwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "e2ff48c6-2c37-4d5a-ae19-84b8362b32bb",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.6"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "n7wwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "4dbaa815-2352-4cc2-bd50-1a5211d973e9",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.7"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "oLwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "4e5ff207-9639-494b-8446-44629221c2e6",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.8"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "obwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "3db33bb3-e726-4bec-9c1f-ed56900e92cc",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.9"
  }
]
```

The REST is:
```ts
/api/lists/items?list_id=ips.txt&value=127.0.0.1/16"
```

This enables the users of lists to quickly add different values and then delete those values using CIDR.

However, for some list types such as `ip_range` you currently cannot delete by query as you can see below as they are more problematic. The "non range" data types you can. We might support this better in the future but this is where we are at the moment with limitations and bugs 

```ts
❯ ./delete_list_item_by_value.sh ip_range.txt 192.168.0.6
{
  "message": "[query_shard_exception] failed to create query: '192.168.0.5-192.168.0.8' is not an IP string literal., with { index_uuid=\"D41WP1pwRa2INSafOfZ0Pw\" & index=\".items-frank-default-000001\" }",
  "status_code": 400
}
❯ ./delete_list_item_by_value.sh ip_range.txt 192.168.0.5-192.168.0.8
{
  "message": "[query_shard_exception] failed to create query: '192.168.0.5-192.168.0.8' is not an IP string literal., with { index_uuid=\"D41WP1pwRa2INSafOfZ0Pw\" & index=\".items-frank-default-000001\" }",
  "status_code": 400
}
```